src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -690,18 +725,51 @@ private void updateSourcePropertiesForScreenDest(JsonNode destContent, SourcePro
 		JsonNode layers = destContent.get(BarcoE2Constant.LAYERS);
 		List<Integer> listSourceIndexes = new ArrayList<>();
 		for (int j = 0; j < layers.size(); j++) {
+			if (layers.get(j).get(BarcoE2Constant.LAST_SRC_IDX) == null) {
```

Comment:
layers.get(j) could be extracted into variable and reused. 
Also, is there a chance for a NPE?

Reply:
Fixed